debug-cli/scripts/overwrite-key/recovery-script.sh

```diff
@@ -0,0 +1,238 @@
+                        BACKUP_COUNT=$(ls -1 "${BROKER_BACKUP_PATH}" 2>/dev/null | grep -v ".checksum" | wc -l)
+                        echo "    ${BACKUP_SIZE} (${BACKUP_COUNT} snapshots)"
+                else
+                        echo "    ERROR: Backup failed for Broker ${broker_id}"
+                        exit 1
+                fi
+        else
+                echo "  WARNING: No snapshots directory for Broker ${broker_id}"
+        fi
+done
+echo "Backups created successfully"
+echo ""
+
+# Step 4: Run cdbg to update key values on leader
+echo "[4/5] Updating key values using cdbg on leader broker ${LEADER_BROKER}..."
+
+# Build the cdbg command
+CDBG_CMD="/usr/local/camunda/bin/cdbg state update-key"
+CDBG_CMD="${CDBG_CMD} --root ${LEADER_PATH}"
+CDBG_CMD="${CDBG_CMD} --runtime ${RUNTIME_PATH}"
+CDBG_CMD="${CDBG_CMD} --snapshot ${SELECTED_SNAPSHOT}"
+CDBG_CMD="${CDBG_CMD} --partition-id ${PARTITION_ID}"
+CDBG_CMD="${CDBG_CMD} --key ${NEW_KEY}"
+[ -n "${NEW_MAX_KEY}" ] && CDBG_CMD="${CDBG_CMD} --max-key ${NEW_MAX_KEY}"
+CDBG_CMD="${CDBG_CMD} --verbose"
+
+# Print the full command being executed
+echo "Running: ${CDBG_CMD}"
+echo ""
+
+eval "${CDBG_CMD}"
+
+echo ""
+echo "Key update completed successfully!"
+echo ""
+
+# Delete the original snapshot since cdbg created a new one
+echo "Removing original snapshot: ${SELECTED_SNAPSHOT}"
+rm -rf "${LEADER_PATH}/snapshots/${SELECTED_SNAPSHOT}"
+# Also remove the checksum file if it exists
+[ -f "${LEADER_PATH}/snapshots/${SELECTED_SNAPSHOT}.checksum" ] && rm -f "${LEADER_PATH}/snapshots/${SELECTED_SNAPSHOT}.checksum"
+echo "Original snapshot removed"
+echo ""
+
+echo "Updated snapshot location: ${LEADER_PATH}/snapshots/"
+ls -la "${LEADER_PATH}/snapshots/"
+echo ""
+
+# Step 5: Copy updated partition data to replicas
+echo "[5/5] Copying updated partition data to replicas..."
+
+for broker_id in $PARTITION_BROKER_IDS; do
+        if [ "$broker_id" != "$LEADER_BROKER" ]; then
+                REPLICA_PATH="/mnt/broker-${broker_id}/raft-partition/partitions/${PARTITION_ID}"
+                echo "  Syncing to Broker ${broker_id}..."
+
+                # Remove all existing snapshots and copy new ones from leader
+                rm -rf "${REPLICA_PATH}/snapshots/"*
+                cp -r "${LEADER_PATH}/snapshots/"* "${REPLICA_PATH}/snapshots/"
+                sync # Ensure replicated snapshots are persisted to disk
+
+                # Verify copy
+                REPLICA_SNAPSHOT_COUNT=$(ls -1 "${REPLICA_PATH}/snapshots/" 2>/dev/null | grep -v ".checksum" | wc -l)
+                echo "    Updated (${REPLICA_SNAPSHOT_COUNT} snapshots)"
+        fi
+done
+
+echo ""
+echo "Ensuring all data is persisted to disk..."
+sync
+echo "All changes flushed to disk"
+echo ""
+echo "============================================="
+echo "=== Recovery Complete ==="
+echo "============================================="
+echo "Partition ${PARTITION_ID} has been recovered with:"
```

Comment:
❓ This means if we have this for multiple partitions we need to rerun this right?